src/python_testing/TC_IDM_2_2.py

```diff
@@ -0,0 +1,596 @@
+        # Verify on the TH that the DUT returns data from the expected attribute path.
+        self.print_step(17, "Read a read request of attribute type enum")
+        await self.check_attribute_read_for_type(MatterIntEnum)
+
+        # Step 18
+        # TH sends the Read Request Message to the DUT to read an attribute of data type bitmap.
+        # If the device does not have an attribute of data type bitmap, skip this step.
+        self.print_step(18, "Read a read request of attribute type bitmap")
+
+        await self.check_attribute_read_for_type(IntFlag)
+
+        # Step 19
+        # TH sends the Read Request Message to the DUT to read any attribute to an unsupported Endpoint.
+        # DUT responds with the report data action.
+        # Verify on the TH that the DUT sends the status code UNSUPPORTED_ENDPOINT
+
+        self.print_step(19, "Send the Read Request Message to the DUT to read any attribute to an unsupported Endpoint")
+        supported_endpoints = set(self.endpoints.keys())
+        all_endpoints = set(range(max(supported_endpoints)+2))
+        unsupported = list(all_endpoints - supported_endpoints)
+        # Read descriptor
+        result = await self.read_single_attribute_expect_error(endpoint=unsupported[0], cluster=Clusters.Descriptor, attribute=Clusters.Descriptor.Attributes.FeatureMap, error=Status.UnsupportedEndpoint)
+
+        # Step 20
+        # TH sends the Read Request Message to the DUT to read any attribute to an unsupported cluster.
+        # DUT responds with the report data action.
+
+        # Verify on the TH that the DUT sends the status code UNSUPPORTED_CLUSTER
+        self.print_step(20, "Send the Read Request Message to the DUT to read any attribute to an unsupported cluster")
+
+        for endpoint_id, endpoint in self.endpoints.items():
+            for cluster_type, cluster in endpoint.items():
+                if global_attribute_ids.cluster_id_type(cluster_type.id) != global_attribute_ids.ClusterIdType.kStandard:
+                    continue
+
+                all_clusters = set(list(ClusterObjects.ALL_CLUSTERS.keys()))
+                dut_clusters = set(list(x.id for x in endpoint.keys()))
+
+                unsupported = [id for id in list(all_clusters - dut_clusters) if global_attribute_ids.attribute_id_type(id)
+                               == global_attribute_ids.AttributeIdType.kStandardNonGlobal]
+
+                if unsupported:
+                    unsupported_attribute = (ClusterObjects.ALL_ATTRIBUTES[unsupported[0]])[0]
+
+                    result = await self.read_single_attribute_expect_error(endpoint=endpoint_id, cluster=ClusterObjects.ALL_CLUSTERS[unsupported[0]], attribute=unsupported_attribute, error=Status.UnsupportedCluster)
+                    asserts.assert_true(isinstance(result.Reason, InteractionModelError),
+                                        msg="Unexpected success reading invalid cluster")
```

Comment:
Can you add an else here to mark this step as skipped in the unlikely even that the dut has no unsupported clusters? (This should honestly never happen, but it's good to flag it at least)

Reply:
Added